src/radical/pilot/agent/radical-pilot-agent-multicore.py

```diff
@@ -2204,6 +2443,381 @@ def construct_command(self, task_exec, task_args, task_numcores,
         return poe_command, None
 
 
+# ==============================================================================
+#
+# The Launch Method Implementation for Running YARN applications
+#
+class LaunchMethodYARN(LaunchMethod):
+
+    # --------------------------------------------------------------------------
+    #
+    def __init__(self, cfg, logger):
+
+        LaunchMethod.__init__(self, cfg, logger)
+
+
+    # --------------------------------------------------------------------------
+    #
+    @classmethod
+    def lrms_config_hook(cls, name, cfg, lrms, logger):
+        """
+        FIXME: this config hook will inspect the LRMS nodelist and, if needed,
+               will start the YRN cluster on node[0].
+        """
+
+        logger.info('Hook called by YARN LRMS with the name %s'%lrms.name)
+
+        def config_core_site(node):
+
+            core_site_file = open(os.getcwd()+'/hadoop/etc/hadoop/core-site.xml','r')
+            lines = core_site_file.readlines()
+            core_site_file.close()
+
+            prop_str  = '<property>\n'
+            prop_str += '  <name>fs.default.name</name>\n'
+            prop_str += '    <value>hdfs://%s:54170</value>\n'%node
+            prop_str += '</property>\n'
+
+            lines.insert(-1,prop_str)
+
+            core_site_file = open(os.getcwd()+'/hadoop/etc/hadoop/core-site.xml','w')
+            for line in lines:
+                core_site_file.write(line)
+            core_site_file.close()
+
+        def config_hdfs_site(nodes):
+
+            hdfs_site_file = open(os.getcwd()+'/hadoop/etc/hadoop/hdfs-site.xml','r')
+            lines = hdfs_site_file.readlines()
+            hdfs_site_file.close()
+
+            prop_str  = '<property>\n'
+            prop_str += ' <name>dfs.replication</name>\n'
+            prop_str += ' <value>1</value>\n'
+            prop_str += '</property>\n'
+
+            prop_str += '<property>\n'
+            prop_str += '  <name>dfs.name.dir</name>\n'
+            prop_str += '    <value>file:///tmp/hadoop/hadoopdata/hdfs/namenode</value>\n'
+            prop_str += '</property>\n'
+
+            prop_str += '<property>\n'
+            prop_str += '  <name>dfs.data.dir</name>\n'
+            prop_str += '    <value>file:///tmp/hadoop/hadoopdata/hdfs/datanode</value>\n'
+            prop_str += '</property>\n'
+
+            lines.insert(-1,prop_str)
+
+            hdfs_site_file = open(os.getcwd()+'/hadoop/etc/hadoop/hdfs-site.xml','w')
```

Comment:
Again, this is a bit difficult. We create our own YARN cluster, which means that we need to set the above paths. They can be parametrized after the /tmp folder.

Is there a specific reason for making them parametric?